tsconfig.base.json

```diff
@@ -1,10 +1,11 @@
 {
-  "extends": "@tsconfig/node18/tsconfig.json",
+  "extends": "@tsconfig/node20/tsconfig.json",
   "compilerOptions": {
     "allowUnreachableCode": false,
     "declaration": true,
     "noImplicitOverride": true,
     "noImplicitReturns": true,
-    "noUnusedParameters": true
+    "noUnusedParameters": true,
+    "isolatedModules": true
```

Comment:
ts-jest had some warnings telling us to set [`isolatedModules`](https://www.typescriptlang.org/tsconfig/#isolatedModules) to "true". This uncovered a few minor issues with the way we were exporting types from a couple of modules.